infrastructure/zk/src/validate_migrations.ts

```diff
@@ -0,0 +1,40 @@
+import * as utils from './utils';
+import fs from 'fs';
+
+async function validateMigration(filePath: string) {
+    let content = await fs.promises.readFile(filePath, { encoding: 'utf-8' });
+    content = content.toUpperCase();
+    const isBackwardsCompatible = !content.includes('RENAME') && !content.includes('DROP');
```

Comment:
As discussed, I don't really like this check. E.g., AFAIU, it will mark [this migration](https://github.com/matter-labs/zksync-era/blob/b947d02c522a30026c28fc915d7fda1b4a50ffac/core/lib/dal/migrations/20240130151508_add_extra_fields_to_snapshot_recovery.up.sql) as false positive:

```sql
-- `miniblock_root_hash` should be renamed to `miniblock_hash`, but we cannot do it straightforwardly
```

(notice the "rename" use). As a comprehensive solution, you'd need to parse SQL (there should be packages for this, right?) and blacklist specific statement types.

Reply:
I don't think it's OK to have a false positive for the explicit opposite of the tested predicate. @RomanBrodetski WDYT?

Reply:
Taking a step back, I'd like to solve it in a more general way, for example running int testds against the old server version with new DB schema.

I think those false positives are indeed a bummer (I can even imagine a case when someone adds migration to the exceptions list because of said comment, and then adds a rename/delete statement without realising it's now breaking - artificial example of course, but still). But also parsing SQL just to see if there is a drop or rename is an overkill.

@tomg10 would having a CI job that tests new schema against old code be sufficient in place of this PR?